pkg/config/analysis/analyzers/externalcontrolplane/externalcontrolplane.go

```diff
@@ -0,0 +1,121 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package externalcontrolplane
+
+import (
+	"fmt"
+	"net"
+	"net/url"
+
+	v1 "k8s.io/api/admissionregistration/v1"
+
+	"istio.io/istio/pkg/config"
+	"istio.io/istio/pkg/config/analysis"
+	"istio.io/istio/pkg/config/analysis/msg"
+	"istio.io/istio/pkg/config/resource"
+	"istio.io/istio/pkg/config/schema/gvk"
+	"istio.io/istio/pkg/slices"
+)
+
+type ExternalControlPlaneAnalyzer struct{}
+
+// Compile-time check that this Analyzer correctly implements the interface
+var _ analysis.Analyzer = &ExternalControlPlaneAnalyzer{}
+
+// Metadata implements Analyzer
+func (s *ExternalControlPlaneAnalyzer) Metadata() analysis.Metadata {
+	return analysis.Metadata{
+		Name:        "externalcontrolplane.ExternalControlPlaneAnalyzer",
+		Description: "Checks that the remote IstioOperator resources reference an external controle plane",
+		Inputs: []config.GroupVersionKind{
+			gvk.ValidatingWebhookConfiguration,
+			gvk.MutatingWebhookConfiguration,
+		},
+	}
+}
+
+// Analyze implements Analyzer
+func (s *ExternalControlPlaneAnalyzer) Analyze(c analysis.Context) {
+	isRemoteCluster := c.Exists(gvk.ValidatingWebhookConfiguration, resource.NewShortOrFullName("", "istio-validator-external-istiod")) && c.Exists(gvk.MutatingWebhookConfiguration, resource.NewShortOrFullName("", "istio-sidecar-injector-external-istiod"))
```

Comment:
I think these may not be hard-coded names, and they are depending on setting?

Reply:
These webhook configurations come from a `istioctl manifest generate -f remote-config-cluster.yaml --set values.defaultRevision=default` call. More on that can be found here: https://istio.io/latest/docs/setup/install/external-controlplane/#set-up-the-remote-config-cluster

So, yes I would agree that _if_ a user were to change the names of these configurations the analyzer would not be able to find the configs and thus won't run. With that said, I do not know of a way to find these specific configs without using their name.